spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -407,21 +412,21 @@ else if (theListener instanceof BatchMessageListener) {
 				this.batchErrorHandler = errHandler == null ? new BatchLoggingErrorHandler()
 						: (BatchErrorHandler) errHandler;
 			}
-			else if (theListener instanceof AcknowledgingMessageListener) {
+			else if (this.theListener instanceof AcknowledgingMessageListener) {
 				this.listener = null;
-				this.acknowledgingMessageListener = (AcknowledgingMessageListener<K, V>) theListener;
+				this.acknowledgingMessageListener = (AcknowledgingMessageListener<K, V>) this.theListener;
 				this.batchListener = null;
 				this.batchAcknowledgingMessageListener = null;
 				this.isBatchListener = false;
 				validateErrorHandler(false);
```

Comment:
Thanks - fixed this in [this PR](https://github.com/spring-projects/spring-kafka/pull/176).